### PR TITLE
Add support for using --use-application-binary in drive command.

### PR DESCRIPTION
**Pull request by nonvachara-paloit**
### Add support for using --use-application-binary in drive command.

As `flutter drive` already supports this, so this commit added a new command line argument `--use-application-binary` in `patrol drive` and pass that to `flutter drive` command.

Directly relates to #65 .

### CODE REVIEW

File: packages/patrol_cli/lib/src/features/drive/drive_command.dart [hunk 1]
```diff
@@ -118,6 +119,14 @@ class DriveCommand extends StagedCommand<DriveCommandConfig> {
         abbr: 'n',
         help: 'Repeat the test n times.',
         defaultsTo: '$_defaultRepeats',
+      )
+      ..addOption(
+        'use-application-binary',
+        help:
+            'Specify a pre-built application binary to use when running. For Android applications, '
+            'this must be the path to an APK. For iOS applications, the path to an IPA. Other device types '
```
Comment (bartekpacia) [inline packages/patrol_cli/lib/src/features/drive/drive_command.dart:127]: Does APP (for iOS) also work?

File: packages/patrol_cli/lib/src/features/drive/test_runner.dart [hunk 2]
```diff
@@ -83,6 +83,15 @@ class TestRunner extends Disposable {
     _repeats = newValue;
   }
 
+  String? _useApplicationBinary;
+  set useApplicationBinary(String? newValue) {
+    if (_running) {
+      throw StateError('repeats cannot be changed while running');
```
Comment (bartekpacia) [inline packages/patrol_cli/lib/src/features/drive/test_runner.dart:89]:
```suggestion
      throw StateError('application binary cannot be changed while running');
```

File: packages/patrol_cli/lib/src/features/drive/test_runner.dart [hunk 3]
```diff
@@ -175,11 +184,13 @@ class TestRunner extends Disposable {
...
-          } catch (_) {
-            targetRuns.add(TargetRunStatus.failedToBuild);
-            continue;
+          if (_useApplicationBinary == null) {
+            try {
+              await builder(target, device);
+            } catch (_) {
+              targetRuns.add(TargetRunStatus.failedToBuild);
+              continue;
+            }
```
Comment (bartekpacia) [inline packages/patrol_cli/lib/src/features/drive/test_runner.dart:193]: make an early return ("_useApplicationBinary != null; then return")

File: packages/patrol_cli/lib/src/features/drive/drive_command.dart (hunk 1 above)
Comment (nonvachara-paloit) [reply packages/patrol_cli/lib/src/features/drive/drive_command.dart:127]: I tried with `--use-application-binary=build/ios/Debug-iphonesimulator/Runner.app`, also work.

File: packages/patrol_cli/lib/src/features/drive/test_runner.dart (hunk 3 above)
Comment (nonvachara-paloit) [reply packages/patrol_cli/lib/src/features/drive/test_runner.dart:193]: I'm not sure we're on the same page here. 🤔
We checked whether _useApplicationBinary (from parseInput() then config object).
If it was specified, then we skip the building part only.
The rest should continue to execute normally.

File: packages/patrol_cli/lib/src/features/drive/drive_command.dart (hunk 1 above)
Comment (bartekpacia) [reply packages/patrol_cli/lib/src/features/drive/drive_command.dart:127]: OK 👍 

Please update the description to state this :)

Comment (nonvachara-paloit) [reply packages/patrol_cli/lib/src/features/drive/drive_command.dart]: A new commit was pushed to state iOS path to a .app in help text.

**Comment by bartekpacia** [comment]
Thanks for your contribution! 💙

**Pull request merged**